message_processor.go

```diff
@@ -50,34 +77,55 @@ func consumeLog(lf *logPair) {
 			logrus.WithField("id", lf.info.ContainerID).WithError(err).Debug("Ignoring error")
 			dec = protoio.NewUint32DelimitedReader(lf.stream, binary.BigEndian, 1e6)
 		}
-		if sendMessage(lf.splunkl, &buf, lf.info.ContainerID) == false {
-			continue
+		// Append to temp buffer
+		appendToTempBuffer(tmpBuf, &buf)
+		// Check for temp buffer timer expiration
+		diff := time.Now().Sub(tempBufferTimer)
+		if diff > tempMsgBufferHoldDuration {
+			tmpBuf.bufferHoldDurationExpired = true
 		}
-		if sendMessage(lf.jsonl, &buf, lf.info.ContainerID) == false {
+
+		if !sendMessage(lf.splunkl, &buf, tmpBuf, lf.info.ContainerID) {
```

Comment:
The logic here is not very clear to me.

if sendMessage returns false, say because of empty string, the code will skip the loop and do not reset buf and tmpBuf. This looks wrong to me as we probably should just remove that message from both buffers.

Reply:
I think the meaning of the return value of sendMessage() has changed here. Before your change, it means whether a log message is put to the next buffer to event batching. Now it is unclear to me what the return value means.